modules/ml/src/gbt.cpp

```diff
@@ -1292,13 +1292,18 @@ CvGBTrees::calc_error( CvMLData* _data, int type, std::vector<float> *resp )
         return -FLT_MAX;
 
     float* pred_resp = 0;
+    bool userAllocated = false;
+
     if (resp)
     {
         resp->resize(n);
         pred_resp = &((*resp)[0]);
     }
     else
+    {
         pred_resp = new float[n];
+        userAllocated = true;
```

Comment:
I would revert the `userAllocated` flag. IMO, `userAllocated = true` means that memory is provided by user, not allocated by us.